### ISSUE TITLE
upset_query to highlight any set that intersects with a given subset of variables

### ISSUE THREAD

**Issue by mathzero** [krassowski/complex-upset#97]
Hi Michael,

Thanks for creating such a great package.

I'm trying to create an upset_query that will check if a given set contains one of a subset of my variables, and if it does, highlights the corresponding intersection matrices and sizes.

In pseudo code, it would be something like

`
        upset_query(
            intersect=c([any one of my subset of variables], [any other variable]'),
            color='red',
            fill='red',
            only_components=c('intersections_matrix', 'Intersection size')
        )
`

Is there a way of achieving this without manually querying all the possible intersections?

Cheers!

ComplexUpset version 1.1

**Comment by krassowski** [comment]
I'm having a slightl difficulty understanding what you are trying to achieve. It would probably help if you could create an example using the `movies` dataset.

**Comment by mathzero** [comment]
OK apologies for the ugly photoshop, but here's what I'm aiming for. Say I want to highlight all the set that contain one or more drama or comedy movies, I would end up with something like this:
<img width="972" alt="Screenshot 2021-02-05 at 23 41 41" src="https://user-images.githubusercontent.com/29785762/107100477-6185ff80-680c-11eb-80d9-6b1d5bb8e223.png">

Does this make sense?

**Comment by krassowski** [comment]
Yes, that's great! There is currently no built-in method to do so, but we could have a function like in: https://github.com/krassowski/complex-upset/issues/96#issuecomment-770058598 for this use case. An alternative would be to add `contains` or `belongs_to` argument to `upset_query`. I wonder how to handle more complex conditions e.g belongs to X but not Z. Maybe it should be `matches` argument taking a string or a function. If it is a string: check if a region (superset) contains a member with that name, if it is a function then use it as a filter for sets.

**Comment by mathzero** [comment]
Ah thanks! That's a good pointer. I'll have a crack at putting something together and post here if I get it working.
